R/register_handlers.R

```diff
@@ -0,0 +1,98 @@
+#' Register handlers for logging messages, warnings and errors
+#'
+#' @param namespace (`character(1)`) the logger namespace
+#' @param package (`character(1)`) the package name
+#'
+#' @return `NULL` invisibly. Called for its side effects.
+#'
+#' @details This function registers global handlers for messages, warnings and errors.
+#' The handlers will investigate the call stack and if it contains a function
+#' from the package, the message, warning or error will be logged into the respective
+#' namespace.
+#' The handlers are registered only once per package and type. Consecutive calls will be ignored.
+#'
+#' @note Registering handlers is disallowed within `tryCatch` blocks which is used throughout
+#' base R and other packages. Because of this, the `register_handlers` function is wrapped in a
+#' `try` block to avoid errors.
+#'
+#' @seealso [globalCallingHandlers()]
+#'
+#' @export
+#'
+#' @examples
+#' \dontrun{
+#' register_logger(namespace = "teal.logger")
+#' register_handlers("teal.logger")
+#' # see the outcome
+#' globalCallingHandlers()
+#' }
+register_handlers <- function(namespace, package = namespace) {
+  try(
+    {
+      register_handler_type(namespace = namespace, package = package, type = "message")
+      register_handler_type(namespace = namespace, package = package, type = "warning")
+      register_handler_type(namespace = namespace, package = package, type = "error")
```

Comment:
```suggestion
      for(type in c('message', 'warning', 'error')) register_handler_type(namespace = namespace, package = package, type = type)
```

Reply:
don't you think it's a little less readable?

Reply:
It spans into two lines in a suggestion on GitHub, but in the code it's one line that is not that bad.
You can also span into two lines with a direct newline 

```r
for(type in c('message', 'warning', 'error')) 
    register_handler_type(namespace = namespace, package = package, type = type)
```

Not that bad and allows you to extend for other types of handlers in the future. Not a strong fan of any. So up to you :)

Reply:
I think I would stay with the current.
Recently I was on a talk and where a guy explained some programming fundamentals. He mentioned something that I very much agree with - if inside a long-living product: that the code should be very _clear_ and easy to understand. He mentioned this when talking about good naming, styling conventions and others. This also applies to writing small functions. Among other benefits like testing, this assures that your code can be easily read as a set of instructions to execute. Coming back to writing functions, it's easier to read N one-liner functions that does something compared to N big/medium sized blocks that does the same thing.
Coming back to this PR - I find it easier and more clear to read three separate func calls compared to a loop over three elements. Despite being a little bit longer and a little repetitive - it gives a reader at the very first glance that this function has three steps.
I hope you are ok with this :)

Reply:
I'm fine. No worries. I wonder if we would change our mind if there were 5 types that we would like to execute registration_handler for in here. Or 10. You can close this convo